README.md

```diff
@@ -1,44 +1,222 @@
+
 ### b
 
+|type(x) == int|type(x) == float|x > 0|First Condition|Second Condition|QUESTION 3 C|
+|--------------|----------------|-----|---------------|----------------|------------|
+|FALSE         |FALSE           |FALSE|FALSE          |FALSE           |FALSE       |
+|FALSE         |FALSE           |TRUE |FALSE          |FALSE           |FALSE       |
+|FALSE         |TRUE            |FALSE|FALSE          |FALSE           |FALSE       |
+|FALSE         |TRUE            |TRUE |FALSE          |FALSE           |FALSE       |
+|TRUE          |FALSE           |FALSE|FALSE          |FALSE           |FALSE       |
+|TRUE          |FALSE           |TRUE |TRUE           |TRUE            |TRUE        |
+|TRUE          |TRUE            |FALSE|TRUE           |TRUE            |TRUE        |
+|TRUE          |TRUE            |TRUE |TRUE           |TRUE            |TRUE        |
+
+```if (type(x) == int and x > 7) or (type(x) == float  and x > 7):```
+
+```if (type(x) == int or type(x) == float) and x > 7:```
+
+```(A AND B) OR (A AND C) = A AND (B OR C)```
+
+The reasons that these are equivalent is because the first statment is the same as saying ```(B and A) OR (C and A)``` which you can rearrange ot make ```(A AND B) OR (A AND C)```.
+
 ### c
 
+|x == 0 |y == 0|First Condition|Second Condition|Question 3 B|
+|-------|------|---------------|----------------|------------|
+|FALSE  |FALSE |TRUE           |TRUE            |TRUE        |
+|FALSE  |TRUE  |TRUE           |TRUE            |TRUE        |
+|TRUE   |FALSE |TRUE           |TRUE            |TRUE        |
+|TRUE   |TRUE  |FALSE          |FALSE           |FALSE       |
+
+``` 
+if x == 0 and y == 0:
+  do_something ()
+else:
+  print("Do Nothing") 
+```
+
+```
+if x != 0 or y != 0:
+  print("Do Nothing")
+else:
+  do_something()
+```
+
+```NOT (A AND B) = NOT A OR NOT B```
+
+The second condition is equal to the identity because you can write it as ```NOT A OR NOT B```.
+
 ### d
 
+|x > 0|x > 10|y > 0|First Condition|Second Condition|QUESTION 3 D|
+|-----|------|-----|---------------|----------------|------------|
+|FALSE|FALSE |FALSE|FALSE          |FALSE           |FALSE       |
+|FALSE|FALSE |TRUE |FALSE          |FALSE           |FALSE       |
+|FALSE|TRUE  |FALSE|FALSE          |FALSE           |FALSE       |
+|FALSE|TRUE  |TRUE |TRUE           |TRUE            |TRUE        |
+|TRUE |FALSE |FALSE|TRUE           |TRUE            |TRUE        |
+|TRUE |FALSE |TRUE |TRUE           |TRUE            |TRUE        |
+|TRUE |TRUE  |FALSE|TRUE           |TRUE            |TRUE        |
+|TRUE |TRUE  |TRUE |TRUE           |TRUE            |TRUE        |
+
+``` 
+if x > 10 or (x > 0 and y > 0):
+  do_something()
+```
+
+```
+if x > 0 and (x > 10 or y > 0)
+  do_something()
+```
+
+```(A OR B) AND (A OR C) = A OR (B AND C)```
+
+The second condition is equal to the identity because you can write the condition as ```A OR (B AND C)```
```

Comment:
Almost -- the conditions here are actually
```
B AND (A OR C) = A OR (B AND C)
```
which is not true in general, but since it is not possible to have `x>10` and not `x>0` the cases in which the expressions disagree are impossible anyway.